tests/integration/cattletest/core/test_container.py

```diff
@@ -172,20 +178,22 @@ def test_container_first_running(admin_client, sim_context):
     assert c.firstRunning == first
 
 
-def test_container_restart(admin_client, sim_context):
-    container = test_container_create_only(admin_client, sim_context)
+def test_container_restart(admin_client, internal_test_client, sim_context):
+    container = test_container_create_only(admin_client, internal_test_client,
+                                           sim_context)
     container = container.start()
-    container = wait_success(admin_client, container)
+    container = internal_test_client.reload(container)
+    container = wait_success(internal_test_client, container)
     _assert_running(container, sim_context)
 
     container = container.restart()
```

Comment:
You are restarting this container as the internal_test_client because you previously reloaded it to assert that it was running.
If it were me, I probably would have reloaded as the admin_client before restart. Testing restart *is* the primary function of this test, so I think that makes sense.

Pendantic @ibuildthecloud ???